pkg/jobs/registry.go

```diff
@@ -654,6 +654,114 @@ func (r *Registry) LoadJobWithTxn(
 	return j, nil
 }
 
+// GetJobInfo fetches the latest info record for the given job and infoKey.
+func (r *Registry) GetJobInfo(
```

Comment:
One thing that's sort of nice about the job writing APIs are that they observe the session claim if there is one (which there is on the job object during execution) and will fail if the job has been preempted. We'll lose this in this PR. I also am rapidly falling out of love with these methods on the registry which take a txn. Can I nudge you towards making a new structure which holds onto a reference to the job and the txn and then has methods to read and write these keys?

maybe like:

```go

func (j *Job) InfoStorage(txn *kv.Txn) InfoStorage

type InfoStorage struct {
    j *Job
    txn *kv.Txn
}

func (is InfoStore) Get(key []byte) (value []byte, exists bool, _ error) 
```